backend-plugin/src/main/java/com/redhat/jenkins/nodesharingbackend/Pool.java

```diff
@@ -114,25 +114,47 @@
     }
 
     public @CheckForNull UsernamePasswordCredentials getExecutorCredential(ExecutorJenkins executor) {
-        if(executor.getCredentialId() != null && !executor.getCredentialId().isEmpty()) {
+        if(executor.getCredentialId() != null) {
+            LOGGER.finest("using credential with id " + executor.getCredentialId() + " for " + executor.getName());
+
             UsernamePasswordCredentials cred = CredentialsMatchers.firstOrNull(
                     CredentialsProvider.lookupCredentials(UsernamePasswordCredentials.class, Jenkins.getInstance(),
                             ACL.SYSTEM, Collections.<DomainRequirement>emptyList()),
                     CredentialsMatchers.withId(executor.getCredentialId())
```

Comment:
FindBugs generates NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE here... (you need to extract the result of call ``executor.getCredentialId()`` to a variable and the use it only... second call can generate NPE obviously by static analysis)

Reply:
Note: Line 118 has the same issue.

Reply:
I missed this report because FindBugs does not fail my local builds. Now it should be ok.